src/Bundle.ts

```diff
@@ -12,68 +12,349 @@ import transformBundle from './utils/transformBundle';
 import collapseSourcemaps from './utils/collapseSourcemaps';
 import callIfFunction from './utils/callIfFunction';
 import error from './utils/error';
-import {
-	dirname,
-	isRelative,
-	isAbsolute,
-	normalize,
-	relative,
-	resolve
-} from './utils/path';
+import { normalize, resolve } from './utils/path';
 import { OutputOptions } from './rollup/index';
 import { RawSourceMap } from 'source-map';
 import Graph from './Graph';
 import ExternalModule from './ExternalModule';
-import ExternalVariable from './ast/variables/ExternalVariable';
 import ExportDefaultVariable from './ast/variables/ExportDefaultVariable';
+import Variable from './ast/variables/Variable';
+import NamespaceVariable from './ast/variables/NamespaceVariable';
+import ExternalVariable from './ast/variables/ExternalVariable';
+import { makeLegal } from './utils/identifierHelpers';
 import { DynamicImportMechanism } from './ast/nodes/Import';
 
+export type BundleDependencies = {
+	id: string;
+	name: string;
+	isBundle: boolean;
+	reexports?: ReexportSpecifier[];
+	imports?: ImportSpecifier[];
+}[];
+
+export type BundleExports = {
+	local: string;
+	exported: string;
+}[];
+
+export interface ReexportSpecifier {
+	reexported: string;
+	imported: string;
+};
+export interface ImportSpecifier {
+	local: string;
+	imported: string;
+};
+
 export default class Bundle {
```

Comment:
I am more and more getting the feeling that `Chunk` might be a better name. Or otherwise, we could start referring to chunks as bundles. At least to me, it seems these are the same concepts. If so, they should not have different names in the code depending on context.

Reply:
Ok I've changed the terminology to:
* Chunk as the name of each group of rendered modules
* Bundle as the name of the group of Chunks for a build